app/src/main/java/com/kms/cura/view/activity/MessageThreadActivity.java

```diff
@@ -200,4 +219,105 @@ public ListView getLvMessage() {
     public boolean isMultiMode() {
         return multiMode;
     }
+
+    private void deleteMessages(final ArrayList<MessageEntity> entities) {
+        new AsyncTask<Object, Void, Void>() {
+            private Exception exception = null;
+            ProgressDialog pDialog;
+
+            @Override
+            protected void onPreExecute() {
+                pDialog = new ProgressDialog(MessageThreadActivity.this);
+                pDialog.setMessage(getString(R.string.loading));
+                pDialog.setCancelable(false);
+                pDialog.show();
+            }
+
+            @Override
+            protected Void doInBackground(Object[] params) {
+                try {
+                    for (MessageEntity entity : entities) {
+                        MessageController.deleteMessage(CurrentUserProfile.getInstance().getEntity(), entity);
```

Comment:
Should get CurrentUserProfile.getInstance().getEntity() and set it to a variable to avoid calling it in a loop like this